server.js

```diff
@@ -4865,6 +4865,150 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+//vscode installs integration
+camp.route(/^\/vscode\/installs\/(.*)\.(svg|png|gif|jpg|json)$/,
+  cache(function (data, match, sendBadge, request) {
+    var repo = match[1];  // eg, `ritwickdey.LiveServer`.
+    var format = match[2];
+    var badgeData = getBadgeData('Installs', data);
+
+    var options = {
+      method: 'POST',
+      url: 'https://marketplace.visualstudio.com/_apis/public/gallery/extensionquery/',
+      headers:
+      {
+        'accept': 'application/json;api-version=3.0-preview.1',
+        'content-type': 'application/json'
+      },
+      body:
+      {
+        filters: [{
+          criteria: [
+            { filterType: 7, value: repo },
+            { filterType: 12, value: '4096' }]
+        }],
+        flags: 914
```

Comment:
What are these values? Could you add a comment?

Are the options identical between these three badges? If so could you make a `createVscodeOptions` to DRY this up?

Reply:
Yep identical! this is for API request. Okay I will separate it. 😃